benchmarks/benchmarks/bench_function_base.py

```diff
@@ -293,3 +308,43 @@ def time_2(self):
 
     def time_2_broadcast(self):
         np.where(self.cond, self.d, 0)
+
+    def time_all_zeros(self):
+        np.where(self.all_zeros)
+
+    def time_random_01_percent(self):
+        np.where(self.rand_cond_01)
+
+    def time_random_20_percent(self):
+        np.where(self.rand_cond_20)
+
+    def time_random_30_percent(self):
+        np.where(self.rand_cond_30)
+
+    def time_random_40_percent(self):
+        np.where(self.rand_cond_40)
+
+    def time_random_50_percent(self):
+        np.where(self.rand_cond_50)
+
+    def time_all_ones(self):
+        np.where(self.all_ones)
+
+    def time_interleaved_zeros_x2(self):
+        np.where(self.rep_zeros_2)
+
+    def time_interleaved_zeros_x4(self):
+        np.where(self.rep_zeros_4)
+
+    def time_interleaved_zeros_x8(self):
+        np.where(self.rep_zeros_8)
+
+    def time_interleaved_ones_x2(self):
+        np.where(self.rep_ones_2)
+
+    def time_interleaved_ones_x4(self):
+        np.where(self.rep_ones_4)
+
+    def time_interleaved_ones_x8(self):
+        np.where(self.rep_ones_8)
```

Comment:
Thanks for those thorough tests!  We could group the `percent` and `interleaved` ones with parameters in `asv` probably, although I am not sure if that doesn't require splitting it into multiple classes.  So fine with me, just wanted to mention it.